### PR TITLE
chore: polish  versions

### PR DESCRIPTION
**Pull request by thxCode**
Now we hug Golang v1.21.

This PR introduces the following changes:
- bump go version to v1.21.
- tidy up golangci-lint configuration.
- polish modules' version.
- fix the changes from the openapi schema lib upgrading.
- use `make deps update` to bump the modules' versions.
- convert the i18n generation to a `pkg/gen` task.

**BREAK CHANGES**
- bump go version to v1.21
- polish modules' version
- openapi schema generation

#1237

### CODE REVIEW

**Review by alexcodelf — APPROVED** [review]
LGTM

File: hack/deps.sh [hunk 1]
```diff
@@ -24,15 +29,15 @@ function dispatch() {
 
   shift 2
   local specified_targets="$*"
-  if [[ -n ${specified_targets} ]] && [[ ! ${specified_targets} =~ ${target} ]]; then
+  if [[ -n ${specified_targets} ]] && [[ ${specified_targets} != "update" ]] && [[ ! ${specified_targets} =~ ${target} ]]; then
```
Comment (gitlawr) [inline hack/deps.sh:32]: How do we use the new update target?

Comment (thxCode) [reply hack/deps.sh:32]: `make deps update` -- upgrade the go modules

**Review by gitlawr — APPROVED** [review]
LGTM

**Pull request merged**